- Add OpenBSD detection
  2ddca0e08c7dbc7a7c794e96c959acc2a84301bb

- Fix build on sparc64 by not erroring out

Index: hwy/detect_compiler_arch.h
--- hwy/detect_compiler_arch.h.orig
+++ hwy/detect_compiler_arch.h
@@ -441,8 +441,6 @@
 #define HWY_ARCH_MAX_BYTES 32
 #elif HWY_ARCH_PPC || HWY_ARCH_S390X || HWY_ARCH_ARM_V7 || HWY_ARCH_ARM_OLD
 #define HWY_ARCH_MAX_BYTES 16
-#else
-#error "Missing case for HWY_ARCH_*"
 #endif
 
 //------------------------------------------------------------------------------
@@ -473,9 +471,16 @@
 #define HWY_OS_FREEBSD 0
 #endif
 
+#if defined(__OpenBSD__)
+#define HWY_OS_OPENBSD 1
+#else
+#define HWY_OS_OPENBSD 0
+#endif
+
 // It is an error to detect multiple OSes at the same time, but OK to
 // detect none of the above.
-#if (HWY_OS_WIN + HWY_OS_LINUX + HWY_OS_APPLE + HWY_OS_FREEBSD) > 1
+#if (HWY_OS_WIN + HWY_OS_LINUX + HWY_OS_APPLE + HWY_OS_FREEBSD +             \
+     HWY_OS_OPENBSD) > 1
 #error "Must not detect more than one OS"
 #endif
 
